lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -804,32 +804,34 @@ static struct nrf_sock_ctx *find_ctx(int fd)
 static void pollcb(struct nrf_pollfd *pollfd)
 {
 	int flags;
-	unsigned int signaled;
 	struct nrf_sock_ctx *ctx;
 
 	ctx = find_ctx(pollfd->fd);
 	if (!ctx) {
 		return;
 	}
 
-	k_poll_signal_check(&ctx->poll, &signaled, &flags);
-	if (!signaled) {
-		flags = 0;
+	if (pollfd->revents & NRF_POLLNVAL) {
+		flags = ZSOCK_POLLNVAL;
+		goto signal;
 	}
 
-	if ((pollfd->events & NRF_POLLIN) && (pollfd->revents & NRF_POLLIN)) {
-		k_poll_signal_raise(&ctx->poll, flags |= ZSOCK_POLLIN);
```

Comment:
have removed these various calls to `k_poll_signal_raise`, now I raise once at the end of the function